src/common/AmazonS3Backend.ts

```diff
@@ -2,15 +2,26 @@ import {Bucket} from './Bucket';
 import {IBackend} from './IBackend';
 
 export class AmazonS3Backend implements IBackend {
+  /**
+   * Amazon S3 API endpoint.
+   */
+  private baseUrl: string;
+
   /**
    * Create an instance of the backend.
+   * @see https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region
    */
-  constructor(public region: string) { }
+  constructor(public region: string) {
+    this.baseUrl = `s3.${region}.amazonaws.com`;
+  }
 
   /**
    * Get a bucket with the given name.
    */
   getBucket(name: string): Bucket {
-    return new Bucket(name);
+    // when the bucket name contains a period ('.') we can't use https
+    // because it treats part of the bucket name as a subdomain and breaks SSL
+    let protocol = name.includes('.') ? 'http' : 'https';
```

Comment:
Should there be a note in the README warning users of the fact that SSL can't be used when a period appears in a bucket name?

Reply:
I actually think the AWS SDK uses a different endpoint for API calls so it doesn't have the same limitation. The exception to this is when I override it for the HTML file uploads. I'll add a note to the README.md for that.